hbase-server/src/test/java/org/apache/hadoop/hbase/master/snapshot/TestSnapshotHFileCleaner.java

```diff
@@ -137,8 +138,15 @@ public void testCorruptedRegionManifest() throws IOException {
     builder.addRegionV2();
     builder.corruptOneRegionManifest();
 
-    fs.delete(SnapshotDescriptionUtils.getWorkingSnapshotDir(rootDir, TEST_UTIL.getConfiguration()),
-      true);
+    long period = Long.MAX_VALUE;
+    SnapshotFileCache cache = new SnapshotFileCache(fs, rootDir, period, 10000000,
+        "test-snapshot-file-cache-refresh", new SnapshotFiles());
+    try {
+      cache.getSnapshotsInProgress();
```

Comment:
Is this testing that we don't throw an exception when the snapshot is corrupted?

Reply:
Going through the context, it tests in case of corrupted snapshot manifest file, it throws out CorruptedSnapshotException and it is handled in getSnapshotsInProgress().

Reply:
In this case, shouldn't the try{} be removed (since it might mask a true failure)?

Reply:
This try{} finally {} does not catch any exception, it is just used for deleting directory created by the test. If getSnapshotsInProgress() throws out any exception, it will fail the test.